datadog_checks_dev/tests/test__env.py

```diff
@@ -44,7 +45,14 @@ def test_environment_run_on_failed_conditions(attempts, expected_call_count):
     condition = mock.MagicMock()
     condition.side_effect = RetryError("error")
 
-    with pytest.raises(RetryError if attempts is None else tenacity.RetryError):
+    expected_exception = tenacity.RetryError
+    if attempts is None:
```

Comment:
I tried to do this in the definition of the test parameters but there running_on_ci() would always evaluate to false